tools/scan_includes.c

```diff
@@ -124,9 +109,37 @@ void scan_file(char *filename) {
 					char *include = xmalloc(length + 1);
 					strncpy(include, buffer, length);
 					include[length] = '\0';
-					printf("%s ", include);
-					if (is_include) {
-						scan_file(include);
+
+					f = fopen(include, "r");
+					char *path = NULL;
+					if (!f) {
+						for (char **include_path = Options.include_paths;
+								include_path < Options.include_paths + Options.include_paths_len;
+								include_path++) {
+							size_t len = strlen(*include_path) + strlen(include) + 1;
+							path = xmalloc(len);
+							snprintf(path, len, "%s%s", *include_path, include);
+							f = fopen(path, "r");
+							if (f) break;
+							free(path);
+							path = NULL;
+						}
+					}
+					if (f) {
+						fclose(f);
+					}
+
+					if (path) {
```

Comment:
at this point you can just `free(include); include = path;` honestly

Reply:
Good idea